filesystem/android/src/main/java/com/capacitorjs/plugins/filesystem/Filesystem.java

```diff
@@ -303,81 +307,123 @@ public void copyRecursively(File src, File dst) throws IOException {
         }
     }
 
-    public JSObject downloadFile(PluginCall call, Bridge bridge, HttpRequestHandler.ProgressEmitter emitter)
-        throws IOException, URISyntaxException, JSONException {
+    public void downloadFile(PluginCall call, Bridge bridge, HttpRequestHandler.ProgressEmitter emitter,
+            FilesystemDownloadCallback callback) throws IOException, URISyntaxException, JSONException {
```

Comment:
I believe this method no longer throws the mentioned exceptions, but rather returns them in `FilesystemDownloadCallback`, therefore we should update the signature

```suggestion
    public void downloadFile(PluginCall call, Bridge bridge, HttpRequestHandler.ProgressEmitter emitter,
            FilesystemDownloadCallback callback) {
```